src/core/thread/data_poll_manager.cpp

```diff
@@ -97,10 +97,14 @@ otError DataPollManager::SendDataPoll(void)
     MeshForwarder &meshForwarder = GetMeshForwarder();
     otError error;
     Message *message;
+    Neighbor *parent;
 
     VerifyOrExit(mEnabled, error = OT_ERROR_INVALID_STATE);
     VerifyOrExit(!meshForwarder.GetNetif().GetMac().GetRxOnWhenIdle(), error = OT_ERROR_INVALID_STATE);
 
+    parent = meshForwarder.GetNetif().GetMle().GetParent();
+    VerifyOrExit((parent != NULL) && parent->IsStateValidOrRestoring(), error = OT_ERROR_INVALID_STATE);
```

Comment:
Yes. I noticed about `GetParent()` but we may change it in future. The check to ensure it is not NULL is definitely safer.